### ISSUE TITLE
If a scenario is saved with a picture puzzle half done, the save file can be loaded but not work

### ISSUE THREAD

**Issue by NicolaZee** [NPBruce/valkyrie#650]
I have a scenario with a custom picture puzzle.
If the scenario is saved and the puzzle is started (but not completed) the save file can be loaded but the scenario cannot be continued.
In addition the main menu stops working.

**Comment by scrubbless** [comment]
I've noticed the main menu not working after saves.

Also a minor one along the save front. If you save the game and need to fix a bug using the editor, the fix is not present and it ruins the save also (and breaks the menu, where you have to force quit).

**Comment by NPBruce** [comment]
The fix shouldn't be present in the save, but it shouldn't break it either.

**Comment by NicolaZee** [comment]
I hope the following will help to track down the bug.

I've knocked together a quick test program consisting of just a tile with a button with the default image puzzle.
Then ran it (without activating the puzzle) and saved.
The save file loaded in and all was fine.
Then ran it and clicked on the button activating the puzzle and saved.
I did not do any puzzle moves.
The save file loaded in but the scenario could not be continued.
So the bug will still occur if a picture puzzle is just looked at and not attempted.
